src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs

```diff
@@ -194,6 +197,29 @@ public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, string>> conf
                 throw new ArgumentException($"Invalid value '{basicAuthSource}' specified for property '{SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource}'");
             }
 
+            if (authenticationHeaderValueProvider != null)
+            {
+                if (username != null || password != null)
+                {
+                    throw new ArgumentException($"Invalid authentication header value provider configuration: Cannot specify both custom provider and username/password");
+                }
+            }
+            else
+            {
+                if (username != null && password == null)
```

Comment:
I've added checks on username/password being null. This didn't exist in the past (no exception thrown when one is null and the other isn't). Not sure if we want to add this going forward, but this should cover all potential "invalid argument" cases

Reply:
could be written slightly neater, but logic lgtm. lgtm.